src/classes/Permission.js

```diff
@@ -45,6 +56,41 @@ export default class Permission {
   }
 
   /**
+   * Requests permission for desktop notifications in a promise based way
+   * @return {Promise}
+   */
+  _requestAsPromise() {
+    const existing = this.get();
+
+    let isGranted = result => (result === this.GRANTED || result === 0);
+
+    /* Permissions already set */
+    var hasPermissions = (existing !== this.DEFAULT);
+
+    /* Safari 6+, Chrome 23+ */
+    var isModernAPI = (this._win.Notification && this._win.Notification.requestPermission);
+
+    /* Legacy webkit browsers */
+    var isWebkitAPI = (this._win.webkitNotifications && this._win.webkitNotifications.checkPermission);
+
+    return new Promise((resolvePromise, rejectPromise) => {
+
+      var resolver = result => (isGranted(result)) ? resolvePromise() : rejectPromise();
+
+      if (hasPermissions) {
+       resolver(existing)
+      }
+      else if (isModernAPI) {
+        this._win.Notification.requestPermission().then(result => { resolver(result) }).catch(rejectPromise)
```

Comment:
Wouldn't this run into the same issue as described in https://github.com/Nickersoft/push.js/issues/117?

The `.then()` implementation is the issue with Safari.

Reply:
I would take a look then and try to fix it. Anyway i should create a separated PR to resolve this specific issue.

Reply:
I'm looking into the fix as well. It was actually me who mistakenly linked the `.then` ticket to this PR. A separate PR does make sense. What you have looks good though